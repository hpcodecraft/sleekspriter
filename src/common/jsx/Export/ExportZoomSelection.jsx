var ExportZoomSelection = React.createClass({
  render: function() {

    var width = 0,
        height = 0;

    switch(this.props.part) {
      case 'spritesheet':
        width = this.props.editor.file.size.width * this.props.editor.frames.x * this.props.zoom;
        height = this.props.editor.file.size.height * this.props.editor.frames.y * this.props.zoom;
        break;

      default:
        width = this.props.editor.file.size.width * this.props.zoom;
        height = this.props.editor.file.size.height * this.props.zoom;
        break;
    }

    return (
      <div>
        <h6>Size</h6>
        <ul>
          <li>
            <input type="range" min={this.props.editor.zoom.min} max={this.props.editor.zoom.max} value={this.props.zoom} onChange={this.setSize} />
            <input type="number" min={this.props.editor.zoom.min} max={this.props.editor.zoom.max} value={this.props.zoom} onChange={this.setSize} />
          </li>
          <li>
            <i>Exported image size: {width}x{height} pixels</i>
          </li>
        </ul>
      </div>
    )
  },
  setSize: function(event) {
    var size = +event.target.value;
    if(size > 50) size = 50;
    if(size < 1) size = 1;
    channel.publish('export.zoom.set', {zoom: size});
  },
});