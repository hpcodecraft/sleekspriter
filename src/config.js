const config = {
  zoom: {
    min: 1,
    max: 50,
  },
  offset: {
    top: 40,
    right: 228,
    bottom: 47, // + 20 for scrollbar
    left: 45,
  },
  fileEncryptionSecret:
    "61d1a52a671076b2f4894be9472cf46e202fa36cd95bd86bacc4e0be7c87a6e2",
};

export default config;
