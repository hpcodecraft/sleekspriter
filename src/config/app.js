const config = {
  defaultFolder: "~",
  defaultName: "unnamed",
  fileEncryptionSecret:
    "61d1a52a671076b2f4894be9472cf46e202fa36cd95bd86bacc4e0be7c87a6e2",
  fileExtension: "pixels",
  offset: {
    top: PLATFORM === "electron" ? 40 : 70,
    right: 228,
    bottom: 27,
    left: 45,
  },
  zoom: {
    min: 1,
    max: 50,
  },
  limits: {
    file: {
      size: {
        height: 64,
        width: 64,
      },
      frames: {
        x: 8,
        y: 8,
      },
    },
  },
};

export default config;
