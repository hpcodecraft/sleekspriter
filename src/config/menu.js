import { findIndex } from "lodash";

import { getPixelsInScope } from "../utils";

const getDefaultMenuConfig = props => {
  const { clipboard, frame, layer, layers, pixels, selection, size } = props;

  const SEPERATOR = { type: "separator" };

  const aboutOption = {
    label: `About ${APPNAME}`,
    click: () => {
      props.modalShow("ModalAbout");
    },
  };

  // -------------------------------------------------------------------------------------------

  const editMenu = {
    label: "Edit",
    screen: ["paint"],
    submenu: [
      {
        label: "Cut",
        accelerator: "CmdOrCtrl+X",
        click: () => {
          const scopedPixels = getPixelsInScope(
            frame,
            layer,
            pixels,
            selection
          );
          props.pixelsCut(frame, layer, scopedPixels, pixels);
        },
      },
      {
        label: "Copy",
        accelerator: "CmdOrCtrl+C",
        click: () => {
          const scopedPixels = getPixelsInScope(
            frame,
            layer,
            pixels,
            selection
          );
          props.pixelsCopy(frame, layer, scopedPixels);
        },
      },
      {
        label: "Paste",
        accelerator: "CmdOrCtrl+V",
        click: () => {
          props.pixelsPaste(frame, layer, clipboard);
        },
      },
      {
        label: "Delete",
        accelerator: "Backspace",
        click: () => {
          const scopedPixels = getPixelsInScope(
            frame,
            layer,
            pixels,
            selection
          );
          props.pixelsDelete(frame, layer, scopedPixels, pixels);
        },
      },
      SEPERATOR,
      {
        label: "Rotate 180°",
        click: () => {
          props.pixelsRotate(frame, layer, size, selection, 180);
        },
      },
      {
        label: "Rotate 90° CW",
        click: () => {
          props.pixelsRotate(frame, layer, size, selection, 90);
        },
      },
      {
        label: "Rotate 90° CCW",
        click: () => {
          props.pixelsRotate(frame, layer, size, selection, -90);
        },
      },
      SEPERATOR,
      {
        label: "Flip Horizontal",
        click: () => {
          props.pixelsFlipHorizontal(frame, layer, size, selection);
        },
      },
      {
        label: "Flip Vertical",
        click: () => {
          props.pixelsFlipVertical(frame, layer, size, selection);
        },
      },
      SEPERATOR,
      {
        label: "Replace color…",
        click: () => {
          props.modalShow("ModalReplaceColor");
        },
      },
      // {label: "Image size…", click: () => { props.modalShow("ModalEditImageSize"); }}
    ],
  };

  // -------------------------------------------------------------------------------------------

  const selectMenu = {
    label: "Select",
    screen: ["paint"],
    submenu: [
      {
        label: "All",
        accelerator: "CmdOrCtrl+A",
        click: () => {
          props.selectionStart({ x: 1, y: 1 });
          props.selectionEnd({ x: size.width, y: size.height });
        },
      },
      {
        label: "Deselect",
        accelerator: "CmdOrCtrl+D",
        click: props.selectionClear,
      },
    ],
  };

  // -------------------------------------------------------------------------------------------

  const layerMenu = {
    label: "Layer",
    screen: ["paint"],
    submenu: [
      {
        label: "Merge with layer above",
        accelerator: "CmdOrCtrl+E",
        click: () => {
          const index = findIndex(layers, { id: layer });
          let layerAbove = null;
          if (layers[index - 1]) {
            layerAbove = layers[index - 1].id;
            props.layerMerge(frame, layerAbove, layer, pixels);
          }
        },
      },
      {
        label: "Merge with layer below",
        accelerator: "CmdOrCtrl+Shift+E",
        click: () => {
          const index = findIndex(layers, { id: props.layer });
          let layerBelow = null;
          if (layers[index + 1]) {
            layerBelow = layers[index + 1].id;
            props.layerMerge(frame, layer, layerBelow, pixels);
          }
        },
      },
    ],
  };

  // -------------------------------------------------------------------------------------------

  const frameMenu = {
    label: "Frame",
    screen: ["paint"],
    submenu: [
      {
        label: "Rotate 180°",
        click: () => {
          props.frameRotate(frame, size, 180);
        },
      },
      {
        label: "Rotate 90° CW",
        click: () => {
          props.frameRotate(frame, size, 90);
        },
      },
      {
        label: "Rotate 90° CCW",
        click: () => {
          props.frameRotate(frame, size, -90);
        },
      },
      SEPERATOR,
      {
        label: "Flip Horizontal",
        click: () => {
          props.frameFlipHorizontal(frame, size);
        },
      },
      {
        label: "Flip Vertical",
        click: () => {
          props.frameFlipVertical(frame, size);
        },
      },
      SEPERATOR,
      {
        label: "Duplicate…",
        click: () => {
          props.modalShow("ModalDuplicateFrame");
        },
      },
    ],
  };

  // -------------------------------------------------------------------------------------------

  const windowMenu = {
    label: "Window",
    screen: ["paint", "export"],
    submenu: [
      {
        label: "Paint",
        accelerator: "CmdOrCtrl+1",
        click: () => {
          props.screenSelect("paint");
        },
      },
      {
        label: "Export",
        accelerator: "CmdOrCtrl+2",
        click: () => {
          props.screenSelect("export");
        },
      },
    ],
  };

  // -------------------------------------------------------------------------------------------

  return {
    aboutOption,
    editMenu,
    selectMenu,
    layerMenu,
    frameMenu,
    windowMenu,
    SEPERATOR,
  };
};

export default getDefaultMenuConfig;
