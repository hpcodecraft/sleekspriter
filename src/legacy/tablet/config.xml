<?xml version="1.0" encoding="UTF-8" ?>
  <widget xmlns   = "http://www.w3.org/ns/widgets"
      xmlns:gap   = "http://phonegap.com/ns/1.0"
      id          = "me.hpcodecraft.@@shortname"
      version     = "@@version" >

  <name>@@name</name>

  <description>
    @@description
  </description>

  <author href="http://hpcodecraft.me" email="mail@hpcodecraft.me">
    @@author
  </author>

  <!-- weinre access -->
  <access origin="*" />

  <feature name="http://api.phonegap.com/1.0/file" />

  <!--<preference name="orientation" value="landscape" />-->
  <preference name="fullscreen" value="true" />
  <!--<preference name="permissions" value="none" />-->

  <gap:plugin name="cordova-plugin-file" />
  <gap:plugin name="cordova-plugin-device" />
</widget>
