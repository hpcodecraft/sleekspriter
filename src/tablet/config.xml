<?xml version="1.0" encoding="UTF-8" ?>
  <widget xmlns   = "http://www.w3.org/ns/widgets"
      xmlns:gap   = "http://phonegap.com/ns/1.0"
      id          = "me.hpcodecraft.@@shortname"
      version     = "@@version" >

  <name>@@name</name>

  <description>
    @@description
  </description>

  <author href="http://hpcodecraft.me" email="mail@hpcodecraft.me">
    @@author
  </author>

  <!-- weinre access -->
  <access origin="*" />

  <preference name="orientation" value="landscape" />
  <preference name="fullscreen" value="true" />
  <!--<preference name="permissions" value="none" />-->
</widget>