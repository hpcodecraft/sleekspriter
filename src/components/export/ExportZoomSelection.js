import React from "react";
import PropTypes from "prop-types";
import { clamp } from "lodash";

import { t } from "../../utils";
import config from "../../config";
import { sizeShape, framesShape } from "../../shapes";

const { max, min } = config.zoom;

const ExportZoomSelection = ({ frames, part, size, zoom, setZoom }) => {
  let text;
  if (part === "spritesheet") {
    text = (
      <i>
        {t("Frame size: ${w}x${h} pixels", {
          w: size.width * zoom,
          h: size.height * zoom,
        })}
        <br />
        {t("Spritesheet size: ${w}x${h} pixels", {
          w: size.width * frames.x * zoom,
          h: size.height * frames.y * zoom,
        })}
        <br />
      </i>
    );
  } else
    text = (
      <i>
        {t("Image size: ${w}x${h} pixels", {
          w: size.width * zoom,
          h: size.height * zoom,
        })}
      </i>
    );

  return (
    <div>
      <h6>{t("Zoom")}</h6>
      <ul>
        <li>
          <input
            type="range"
            min={min}
            max={max}
            value={zoom}
            onChange={e => setZoom(clamp(+e.target.value, min, max))}
          />
          <input
            type="number"
            min={min}
            max={max}
            value={zoom}
            onChange={e => setZoom(clamp(+e.target.value, min, max))}
          />
        </li>
        <li>{text}</li>
      </ul>
    </div>
  );
};

ExportZoomSelection.propTypes = {
  frames: framesShape.isRequired,
  part: PropTypes.string.isRequired,
  size: sizeShape.isRequired,
  zoom: PropTypes.number.isRequired,
  setZoom: PropTypes.func.isRequired,
};

export default ExportZoomSelection;
