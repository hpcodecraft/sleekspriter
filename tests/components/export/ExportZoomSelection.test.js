import config from "config";
import ExportZoomSelection from "components/export/ExportZoomSelection";

describe("ExportZoomSelection", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      frames: {
        x: 2,
        y: 2
      },
      part: "frame",
      size: {
        x: 32,
        y: 32
      },
      zoom: 1,
      setZoom: jest.fn()
    };
    wrapper = shallow(<ExportZoomSelection {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("range input", () => {
    it("fires setZoom handler on change", () => {
      wrapper
        .find("input[type=\"range\"]")
        .simulate("change", { target: { value: 5 } });
      expect(props.setZoom).toHaveBeenCalledWith(5);
    });

    it("clamps the minimum zoom value to 1", () => {
      wrapper
        .find("input[type=\"range\"]")
        .simulate("change", { target: { value: -1 } });
      expect(props.setZoom).toHaveBeenCalledWith(1);
    });

    it(`clamps the maximum zoom value to ${
      config.zoom.max
    } (config.zoom.max)`, () => {
      wrapper
        .find("input[type=\"range\"]")
        .simulate("change", { target: { value: config.zoom.max + 10 } });
      expect(props.setZoom).toHaveBeenCalledWith(config.zoom.max);
    });
  });

  describe("number input", () => {
    it("fires setZoom handler on change", () => {
      wrapper
        .find("input[type=\"number\"]")
        .simulate("change", { target: { value: 5 } });
      expect(props.setZoom).toHaveBeenCalledWith(5);
    });

    it("clamps the minimum zoom value to 1", () => {
      wrapper
        .find("input[type=\"number\"]")
        .simulate("change", { target: { value: -1 } });
      expect(props.setZoom).toHaveBeenCalledWith(1);
    });

    it(`clamps the maximum zoom value to ${
      config.zoom.max
    } (config.zoom.max)`, () => {
      wrapper
        .find("input[type=\"number\"]")
        .simulate("change", { target: { value: config.zoom.max + 10 } });
      expect(props.setZoom).toHaveBeenCalledWith(config.zoom.max);
    });
  });
});
