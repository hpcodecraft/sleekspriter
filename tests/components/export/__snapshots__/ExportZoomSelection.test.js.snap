// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportZoomSelection should render correctly 1`] = `
<div>
  <h6>
    Zoom
  </h6>
  <ul>
    <li>
      <input
        max="50"
        min="1"
        type="range"
        value="1"
      />
      <input
        max="50"
        min="1"
        type="number"
        value="1"
      />
    </li>
    <li>
      <i>
        Image size: \${w}x\${h} pixels
      </i>
    </li>
  </ul>
</div>
`;
